lib/std/os.zig

```diff
@@ -455,7 +455,44 @@ pub fn abort() noreturn {
     if (!builtin.link_libc and builtin.os.tag == .linux) {
         raise(SIG.ABRT) catch {};
 
-        // TODO the rest of the implementation of abort() from musl libc here
+        // If there was a SIGABRT handler installed and it returned, or if
+        // SIGABRT was blocked or ignored, take an Async-Signal-safe lock to
+        // prevent sigaction from installing a new SIGABRT handler, uninstall
+        // any handler that may be present, and re-raise the signal to generate
+        // the default action of abnormal termination.
+        // TODO handle failures on setting sigprocmask
+        _ = linux.sigprocmask(SIG.BLOCK, &linux.all_mask, null);
+
+        // TODO decide where (conditionally) abort_lock mutex should go
+        // - start.zig?
+        // - also decide, if locking logic needs adjustments for this use case
+
+        //var abort_mutex = std.Thread.Mutex;
+        // This lock is intentionally never released.
+        //abort_mutex.lock();
+
+        const sigact = Sigaction{
+            .handler = .{ .sigaction = SIG.DFL },
+            .mask = undefined,
+            .flags = undefined,
+            .restorer = undefined,
+        };
+        _ = linux.sigaction(SIG.ABRT, &sigact, null);
+
+        // TODO musl uses __get_tp() and optional offset into TLS block (pthread_impl.h)
+        _ = linux.tkill(linux.gettid(), SIG.ABRT);
```

Comment:
`Thread,getCurrentId()` should do the trick here.

Reply:
Unfortunately do the types mismatch, because this is how getCurrentId is defined:
```rust
/// Represents a unique ID per thread.
pub const Id = u64;
/// Returns the platform ID of the callers thread.
/// Attempts to use thread locals and avoid syscalls when possible.
pub fn getCurrentId() Id {
    return Impl.getCurrentId();
}
```
and using it for the Kernel requires `intCast` back to the original type, which got lost in the `@bitCast(u32, os_gettidcall())`.

Reply:
@kprotty ideas? feel free to take over this PR.

Reply:
I'll mark this as resolved for now. I did add a TODO for solving this, which can be optimized later and is no regression to status quo.